### PR TITLE
refactor hubpa-projects

### PR DESCRIPTION
**Pull request by pasqualedevita**

### CODE REVIEW

File: azure-devops/projects/hub-pa-projects/hub-pa-api.tf [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
 
 # code review
 resource "azuredevops_build_definition" "hub-pa-api-code-review" {
-  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-ro, azuredevops_project.project]
+  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-pr, azuredevops_project.project]
 
   project_id = azuredevops_project.project.id
-  name       = "code-review"
-  path       = local.build_path
+  name       = "${var.hub-pa-api.repository.name}.code-review"
```
Comment (uolter) [inline azure-devops/projects/hub-pa-projects/hub-pa-api.tf:22]: Not blocking ... I will use the built in function format .... it's more terraform > 0.12

**Review by uolter — DISMISSED** [review]

File: azure-devops/projects/hub-pa-projects/hub-pa-api.tf [hunk 2]
```diff
@@ -1,29 +1,37 @@
-locals {
-  repository_name           = "hub-pa-api"
-  repository_repo_id        = "pagopa/${local.repository_name}"
-  repository_branch_name    = "main"
-  repository_pipelines_path = ".devops"
-  build_path                = "\\${local.repository_name}"
+variable "hub-pa-api" {
```
Comment (GiovanniMancini) [inline azure-devops/projects/hub-pa-projects/hub-pa-api.tf:1]: @pasqualedevita Please note that the project is based on two different repositories. One for backend microservices, which is he one configured here; and a different one for frontend, which is meant to be build into static files and deployed to blob storage + cdn. You can find it [here](https://github.com/pagopa/hub-pa-fe)

**Review by GiovanniMancini — DISMISSED** [review]

File: azure-devops/projects/hub-pa-projects/hub-pa-api.tf (hunk 2 above)
Comment (pasqualedevita) [reply azure-devops/projects/hub-pa-projects/hub-pa-api.tf:1]: yes, we need to create a terraform file for each repository

Comment (pasqualedevita) [reply azure-devops/projects/hub-pa-projects/hub-pa-api.tf:1]: @GiovanniMancini I think that azure devpos pipeline defined here has legacy implementation https://github.com/pagopa/hub-pa-fe/tree/main/.devops
U can take more updated pipeline here https://github.com/pagopa/cgn-onboarding-portal-frontend/tree/master/.devops

File: azure-devops/projects/hub-pa-projects/hub-pa-api.tf (hunk 1 above)
Comment (pasqualedevita) [reply azure-devops/projects/hub-pa-projects/hub-pa-api.tf:22]: thanks Walter, do u mean to use this format?

```suggestion
  name       = format("%s.code-review", var.hub-pa-api.repository.name)
```

**Comment by pasqualedevita** [comment]
pr trigger started, all service connection created to point to TTDIO

**Review by uolter — APPROVED** [review]

**Pull request merged**